js/main.js

```diff
@@ -30,12 +50,17 @@ function handleFormSubmit(event) {
     notes: $form.notes.value,
     entryID: data.nextEntryId
   };
-  data.entries.push(entry);
+  if(data.editing != null) {
+    data.entries[data.editing] = entry;
+    data.editing = null;
+  } else {
+    data.nextEntryId++;
+    data.entries.push(entry);
+  }
```

Comment:
I don't see anything in your submit function that is generating a DOM tree for the new entry, does the new entry show on the entries list when you are navigated there without needing to refresh the page?

Reply:
Yes.